### PR TITLE
[DOC] Minor documentation changes

### PR DESCRIPTION
**Pull request by jmigueprieto**
Pull Request type
----

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Other (please describe): Documentation

Changes in this PR
----

Minor documentation changes to help new devs in getting things up & running.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ Conductor Server is a [Spring Boot](https://spring.io/projects/spring-boot) proj
 #### Development
 The server can be started locally by running `./gradlew bootRun` from the project root. This will start up Conductor with an in-memory persistence and queue implementation. It is not recommended for production use but can come in handy for quickly evaluating what Conductor's all about. For actual production use-cases, please use one of the supported persistence and queue implementations.
 
-You can verify the development server is up by navigating to `http://localhost:8080/swagger-ui/index.html` in a browser.
```
Comment (jmigueprieto) [inline README.md:31]: http://localhost:8080/swagger-ui/index.html opens Swagger UI with the sample PetStore API specification which might be a bit confusing.

File: docs/docs/architecture.md [hunk 2]
```diff
@@ -23,27 +23,26 @@ git clone git@github.com:Netflix/conductor.git
 
 #### Start Local Server
 
-The server is in the directory `conductor/server`.
+The server is in the directory `conductor/server`. To start it execute the following command in the root of the project.
 
 ```shell
-cd server
```
Comment (jmigueprieto) [inline docs/docs/architecture.md:29]: Executing these commands gives the following error
```
FAILURE: Build failed with an exception.

* What went wrong:
Task 'server' not found in project ':conductor-server'.
```

I believe it makes sense to change the instructions to just `./gradlew bootRun`

File: docs/docs/architecture.md [hunk 3]
```diff
@@ -23,27 +23,26 @@ git clone git@github.com:Netflix/conductor.git
...
-../gradlew server
+./gradlew bootRun
 # wait for the server to come online
 ```
-Swagger APIs can be accessed at [http://localhost:8080/](http://localhost:8080/)
```
Comment (jmigueprieto) [inline docs/docs/architecture.md:33]: minor detail, not the correct Swagger UI url even though there's a link on the page returned by http://localhost:8080.

File: docs/docs/configuration/taskdef.md [hunk 4]
```diff
@@ -21,10 +21,12 @@ Conductor maintains a registry of worker tasks.  A task MUST be registered befor
   "timeoutPolicy": "TIME_OUT_WF",
   "retryLogic": "FIXED",
   "retryDelaySeconds": 600,
-  "responseTimeoutSeconds": 3600,
```
Comment (jmigueprieto) [inline docs/docs/configuration/taskdef.md:24]: `"responseTimeoutSeconds": 3600` gives a validation error. It should be `1200` or less.

`ownerEmail` seems to be mandatory now. Relates to https://github.com/Netflix/conductor/issues/2542

**Review by apanicker-nflx — APPROVED** [review]

**Pull request merged**